provider/openstack/local_test.go

```diff
@@ -1854,14 +1891,17 @@ func (s *localHTTPSServerSuite) TestCanBootstrap(c *gc.C) {
 	defer restoreFinishBootstrap()
 
 	// For testing, we create a storage instance to which is uploaded tools and image metadata.
-	metadataStorage := openstack.MetadataStorage(s.env)
-	url, err := metadataStorage.URL("")
+	toolsMetadataStorage := openstack.MetadataStorage(s.env)
+	url, err := toolsMetadataStorage.URL("")
 	c.Assert(err, jc.ErrorIsNil)
 	c.Logf("Generating fake tools for: %v", url)
-	envtesting.UploadFakeTools(c, metadataStorage, s.env.Config().AgentStream(), s.env.Config().AgentStream())
-	defer envtesting.RemoveFakeTools(c, metadataStorage, s.env.Config().AgentStream())
-	openstack.UseTestImageData(metadataStorage, s.cred)
-	defer openstack.RemoveTestImageData(metadataStorage)
+	envtesting.UploadFakeTools(c, toolsMetadataStorage, s.env.Config().AgentStream(), s.env.Config().AgentStream())
+	defer envtesting.RemoveFakeTools(c, toolsMetadataStorage, s.env.Config().AgentStream())
+
+	imageMetadataStorage := openstack.ImageMetadataStorage(s.env)
```

Comment:
The test started to fail because it couldn't find the images to bootstrap.  Not sure of the cause,  but the image metadata should have been in the openstack.ImageMetadataStorage() looking at the other tests.